### PR TITLE
Add addresses from Identify to Connection Pool

### PR DESCRIPTION
**Pull request by folex**

### CODE REVIEW

File: connection-pool/src/behaviour.rs [hunk 1]
```diff
@@ -48,19 +48,58 @@ type SwarmEventType = libp2p::swarm::NetworkBehaviourAction<
...
-    /// Storing addresses of connecting peers to return them in `addresses_of_peer`, so libp2p
-    /// can ask ConnectionPool about these addresses after `DialPeer` is issued
-    Dialing(HashSet<Multiaddr>, Vec<OneshotOutlet<bool>>),
+// #[derive(Debug)]
+// enum Peer {
+//     Connected(HashSet<Multiaddr>),
+//     /// Storing addresses of connecting peers to return them in `addresses_of_peer`, so libp2p
+//     /// can ask ConnectionPool about these addresses after `DialPeer` is issued
+//     Dialing(HashSet<Multiaddr>, Vec<OneshotOutlet<bool>>),
+// }
```
Comment (justprosh) [inline connection-pool/src/behaviour.rs:57]: Remove or leave any comment

File: connection-pool/src/behaviour.rs [hunk 2]
```diff
@@ -232,35 +283,28 @@ impl ConnectionPoolBehaviour {
...
-                            }
-                        }
-                    }
-                };
+                let peer = entry.get_mut();
+                peer.dialing.remove(&maddr);
+                peer.discovered.remove(&maddr);
+                peer.connected.insert(maddr.clone());
+
+                let dial_promises = std::mem::replace(&mut peer.dial_promises, <_>::default());
```
Comment (kmd-fl) [inline connection-pool/src/behaviour.rs:295]: Maybe, you want to use [take](https://doc.rust-lang.org/std/mem/fn.take.html)?

File: connection-pool/src/behaviour.rs [hunk 3]
```diff
@@ -286,54 +330,61 @@ impl ConnectionPoolBehaviour {
...
+
+        let _: Option<()> = try {
+            let peer_id = peer_id?;
+            let contact = self.contacts.get_mut(peer_id)?;
+
+            contact.connected.remove(addr);
+            contact.discovered.remove(addr);
+            contact.dialing.remove(addr);
+            if contact.dialing.is_empty() {
+                let dial_promises = std::mem::replace(&mut contact.dial_promises, <_>::default());
```
Comment (kmd-fl) [inline connection-pool/src/behaviour.rs:376]: Here too (see the comment above).

**Review by kmd-fl — APPROVED** [review]

**Review by justprosh — APPROVED** [review]
I think it will be good to add a simple test for the new builtin in particle-node/tests/kademlia.rs

**Pull request merged**